Java_observer/Observer_patten.java

```diff
@@ -0,0 +1,73 @@
+import java.util.ArrayList;
+import java.util.List;
+
+// Subject interface which store list
+interface StockMarket {
+    void addObserver(Investor investor);
+    void removeObserver(Investor investor);
+    void notifyObservers();
+}
+
+// Observer interface
+interface Investor {
+    void update(String stock, double price);
+}
+
+// Concrete Subject
+class StockMarketImpl implements StockMarket {
+    private List<Investor> investors = new ArrayList<>();
+    private String stock;
+    private double price;
+
+    @Override
+    public void addObserver(Investor investor) {
+        investors.add(investor);
+    }
+
+    @Override
+    public void removeObserver(Investor investor) {
+        investors.remove(investor);
+    }
+
+    @Override
+    public void notifyObservers() {
+        for (Investor investor : investors) {
+            investor.update(stock, price);
+        }
+    }
+
+    public void setStockPrice(String stock, double price) {
+        this.stock = stock;
+        this.price = price;
+        notifyObservers();
+    }
+}
+
+// Concrete Observer
+class InvestorImpl implements Investor {
+    private String name;
+
+    public InvestorImpl(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public void update(String stock, double price) {
+        System.out.println(name + " received update - " + stock + " price: " + price);
+    }
+}
+
+//Main Class(Usage example)
+public class Observer_patten  {
```

Comment:
Doesn't Java prefers to use camelCase instead of snake_case? `Java === community && developers`

Reply:
Yes, you are correct. In Java, the convention is to use camelCase for naming variables, methods, and classes, rather than snake case. 

CamelCase involves starting each word in a variable name with a capital letter, except for the first word, which starts with a lowercase letter. For example 
int myVariable;  
void myMethod(); 
class MyClass;

Snake_case, on the other hand, involves separating words with underscores and using all lowercase letters. 
For example: int my_variable;
void my_method();
class my_class;

Thankyou for bringing my attention to this .I already change in my file